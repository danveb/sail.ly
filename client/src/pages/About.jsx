import { aboutSailing } from "../constants/about"; 
import aboutSailTop from "../assets/about/about.jpg"; 
import "../styles/About.css"; 

export const About = () => {
    return (
        <div className="about">
            <div className="about__wrapper">
                <div className="about__main">
                    <div className="about__top">
                        <img src={aboutSailTop} alt="sailing yacht in ocean" />
                    </div>
                    <div className="about__bottom">
                        {aboutSailing.map((item) => (
                            <div key={item.id}>
                                <h2>{item.header}</h2>
                                <p>{item.text_intro}</p>
                                <p>{item.text_main}</p>
                            </div>
                        ))}
                    </div>
                </div>
            </div>
        </div>
    )
}
