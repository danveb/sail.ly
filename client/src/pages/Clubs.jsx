// import { ClubList } from "../components/ClubList";
import { clubsIntro } from "../constants/clubs";
import clubIntro from "../assets/clubs/club-intro.jpg"; 
import "../styles/Clubs.css"; 
import { ClubList } from "../components/ClubList";

export const Clubs = () => {
    return (
        <>
            <div className="clubs">
                <div className="clubs__wrapper">
                    <div className="clubs__main">
                        <div className="clubs__top">
                            <img src={clubIntro} alt="sailing yacht in ocean" />
                        </div>
                        <div className="clubs__bottom">
                            {clubsIntro.map((item) => (
                                <div key={item.id}>
                                    <h2>{item.header}</h2>
                                    <p>{item.text_intro}</p>
                                    <p>{item.text_main}</p>
                                </div>
                            ))}
                        </div>
                    </div>
                </div>
            </div>
            <ClubList />
        </>
    )
}