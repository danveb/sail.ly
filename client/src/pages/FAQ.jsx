import { FAQItem } from "../components/FAQItem";
import faq from "../assets/faq/faq.jpg"; 
import "../styles/FAQ.css"; 

export const FAQ = () => {
    return (
        <div className="faq">
            <div className="faq__wrapper">
                <div className="faq__main">
                    <div className="faq__top">
                        <img src={faq} alt="sailing yacht in ocean" />
                    </div>
                    <div className="faq__bottom">
                        <FAQItem />
                    </div>
                </div>
            </div>
        </div>
    )
}