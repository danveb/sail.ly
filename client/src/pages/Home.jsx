import chevronDown from "../assets/home/chevron-down.svg"; 
import "../styles/Home.css"; 

export const Home = () => {
    return (
        <div>
            <div className="home">
                <div className="home__top">
                    <div className="home__top--title">
                        <h1>#SS | 23</h1>
                        <h3>Dream in open waters</h3>
                        {/* LINK to clubs */}
                    </div>
                    <div className="home__top--bg"></div>
                    <img src={chevronDown} alt="chevron down arrow" />
                </div>
            </div>
        </div>
    )
}