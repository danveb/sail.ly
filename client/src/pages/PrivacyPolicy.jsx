import "../styles/PrivacyPolicy.css"; 

export const PrivacyPolicy = () => {
    return (
        <div className="privacy">
            <div className="privacy__wrapper">
                <div className="privacy__main">
                    <div className="privacy__top" data-testid="privacy__top">
                        <h1>Privacy Policy</h1>
                        <p>At SAIL.LY we take your privacy seriously. This Privacy Policy explains how we collect, use, and disclose your personal information when you use our web app.</p>
                    </div>
                    <div className="privacy__bottom">
                        <h2>Information We Collect</h2>
                        <p>When you use SAIL.LY we do not collect any of the following personal information:</p>
                        <ul>
                            <li>Contact Information: Name and email address</li>
                            <li>Usage Information: Information about how you use our web app, such as the pages you visit and the features you use</li>
                            <li>Device Information: Information about the device you use to access our web app, such as your IP address, browser type, and operating system</li>
                        </ul>
                        <h2>How We Use Your Information</h2>
                        <p>We do not use your personal information to provide and/or improve our web app. This app is built and designed for anyone that may want to learn more about climatic conditions pertaining to sailing.</p>
                        <h2>Data Retention</h2>
                        <p>We do not retain any personal information simply because it is not needed.</p>
                        <h2>Changes to This Privacy Policy</h2>
                        <p>We may update this Privacy Policy if needed. If we make any material changes we will post a notice on our web app prior to the effective date of the changes.</p>
                    </div>
                </div>
            </div>
        </div>
    )
}