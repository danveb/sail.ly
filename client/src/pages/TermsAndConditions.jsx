import "../styles/TermsAndConditions.css"; 

export const TermsAndConditions = () => {
    return (
        <div className="terms" data-testid="terms">
            <div className="terms__wrapper">
                <div className="terms__main">
                    <div className="terms__top">
                        <h1>Terms And Conditions</h1>
                        <p>Welcome to SAIL.LY! Please read these Terms and Conditions carefully before using our web app. By accessing or using SAIL.LY, you agree to be bound by these Terms and Conditions. If you do not agree to these Terms and Conditions, please do not use our web app.</p>
                    </div>
                    <div className="terms__bottom">
                        <h2>Use of SAIL.LY</h2>
                        <p>SAIL.LY is an application intended for educational purposes. It serves anyone who has little-to-no experience up to experienced sailors.</p>
                        <h2>Intellectual Property</h2>
                        <p>You agree that all materials, products, and services provided on this payment application are the property of SAIL.LY, its affiliates, directors, officers, employees, agents, suppliers, or licensors including all copyrights, trade secrets, trademarks, patents, and other intellectual property.</p>
                        <h2>Applicable Law</h2>
                        <p>By visiting this website, you agree that the laws of the United States of America, without regard to principles of conflict laws, will govern these terms and conditions, or any dispute of any sort that might come.</p>
                    </div>
                </div>
            </div>
        </div>
    )
}