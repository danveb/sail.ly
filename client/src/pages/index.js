export { default as About } from "./About";
export { default as Clubs } from "./Clubs";
export { default as FAQ } from "./FAQ";
export { default as Home } from "./Home";
export { default as PrivacyPolicy } from "./PrivacyPolicy";
export { default as TermsAndConditions } from "./TermsAndConditions"; 