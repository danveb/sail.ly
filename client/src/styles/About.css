.about {
    background-color: #fff; 
}

.about__wrapper {
    max-width: 1200px; 
    height: 100%; 
    margin: 0 auto; 
}

.about__main {
    display: flex; 
    flex-direction: column; 
    justify-content: center; 
    align-items: center; 
    max-width: 590px; 
    margin: auto; 
}

.about__top {
    display: flex; 
}

.about__top img {
    width: 100%; 
    height: 100%; 
    object-fit: contain; 
}

.about__bottom {
    margin: 30px 0px; 
}

@media screen and (max-width: 591px) {
    .about__bottom {
        margin: auto; 
        padding: 25px; 
    }
}

.about__bottom h2 {
    font-size: 24px; 
    font-family: "Prata", sans-serif; 
    margin: 15px 0; 
}

.about__bottom p {
    font-family: "Yrsa", sans-serif; 
    color: #333; 
    margin: 15px 0; 
}