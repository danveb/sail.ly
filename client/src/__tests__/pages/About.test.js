import { render, screen } from "@testing-library/react";
import { About } from "../../pages/About"; 

describe("About Component", () => {
    // smoke test
    test("renders without crashing", () => {
        render(<About />); 
    });

    // snapshot test 
    test("matches snapshot", () => {
        const { asFragment } = render(<About />); 
        expect(asFragment()).toMatchSnapshot(); 
    });

    // getByAltText 
    test("img tag has attribute for screen readers", () => {
        render(<About />); 
        const altText = screen.getByAltText("sailing yacht in ocean"); 
        expect(altText).toBeInTheDocument(); 
    });

    // getByText h2 
    test("h2 tag correctly renders text", () => {
        render(<About />); 
        const textElement = screen.getByText("Why"); 
        expect(textElement).toBeInTheDocument(); 
        expect(textElement).not.toBe("Hello World"); 
    });

    // getByText p
    test("p tag renders text-intro", () => {
        render(<About />); 
        const textElement = screen.getByText("Sailing allows one to escape from the hustle and bustle", { exact: false }); 
        expect(textElement).toBeInTheDocument(); 
    });
}); 