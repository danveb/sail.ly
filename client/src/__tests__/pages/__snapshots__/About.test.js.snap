// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`About Component matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="about"
  >
    <div
      class="about__wrapper"
    >
      <div
        class="about__main"
      >
        <div
          class="about__top"
        >
          <img
            alt="sailing yacht in ocean"
            src="about.jpg"
          />
        </div>
        <div
          class="about__bottom"
        >
          <div>
            <h2>
              Why
            </h2>
            <p>
              Sailing allows one to escape from the hustle and bustle of everyday life and immerse in the beauty of nature. From the sensation of the wind, the sound of the waves, and the sun shining brightly create a sense of freedom and tranquility hard to replicate.
            </p>
            <p>
              Sailing provides a unique opportunity to bond with nature and create life-lasting memories throughout each journey. No matter the level of expertise, from beginner to a seasoned sailor, one becomes confident and energized to travel on the waters.
            </p>
          </div>
          <div>
            <h2>
              Wind
            </h2>
            <p>
              Wind is an important factor when sailing because it provides the power necessary to move the sailboat through the water. The wind's force on the sail generates a forward thrust that propels the boat forward. The sailboat's speed and direction are determined by the wind direction and the angle of the sail relative to the wind. A sailor must understand how to use the wind to their advantage in order to control the boat's speed and direction.
            </p>
            <p>
              For example, by adjusting the sail's angle and shape, a sailor can harness the wind's energy to achieve maximum speed, or change direction by tacking or jibing. Understanding how to read and respond to changes in wind direction and speed is essential for a sailor to navigate safely and efficiently. In addition to propulsion, wind also affects a sailboat's stability and maneuverability. Strong winds can make the boat unstable, and gusts of wind can cause the boat to heel (tilt) and change direction suddenly. Sailors must be skilled at handling the boat in different wind conditions and adjusting their sail plan accordingly.
            </p>
          </div>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
